gourmet/defaults/defaults.py

```diff
@@ -44,31 +45,19 @@
     # 'capitalisedNouns' means that you don't want to use lower() anyway, cos it's
     #  ungramatical e.g. in the german Language, Nouns are written with Capital-Letters.
 
-# NOW WE DO AUTOMATED STUFF
-def add_itm (kd, k, v):
-    if k in kd:
-        kd[k].append(v)
-    else:
-        kd[k]=[v]
-
 ## now we set up our dictionaries
-lang.keydic = {}
-lang.shopdic = {}
-for lst in lang.SYNONYMS:
-    k = lst[0]
-    for i in lst:
-        add_itm(lang.keydic,k,i)
-
-for amb,lst in list(lang.AMBIGUOUS.items()):
-    if amb in lang.keydic:
-        lang.keydic[amb] += lst
-    else:
-        lang.keydic[amb] = lst
+lang.keydic = defaultdict(list)
+for variants in lang.SYNONYMS:
+    preferred = variants[0]
+    lang.keydic[preferred].extend(variants)
+
+for preferred, alternatives in lang.AMBIGUOUS.items():
+    lang.keydic[preferred].extend(alternatives)
+
+for itemname, key, _ in lang.INGREDIENT_DATA:
+    lang.keydic[key].append(itemname)
```

Comment:
Should there be a final step to convert `keydic` back to a regular dict? I haven't looked at where it's used, but code may use a distinction between something that's not in the dict at all and something that's there with an empty list.

Reply:
I am not so sure. There's not `except KeyError` in the codebase. Also I think the semantics fit, as this is a mapping from a key to alternative variants, synonyms, etc

If I plug in something that is not a key for anything, I get no alternatives, etc.

I know, it's kinda stupid as this Mapping should probably be flipped around, putting in some variant that maps to a key. And then a `KeyError` would probably be appropriate, if that is not an alternative that maps to a key.

Reply:
Fair enough, sounds reasonable to leave it as a defaultdict for now. It makes sense to leave things like inverting dictionaries until it's basically working.